homeassistant/components/todoist/calendar.py

```diff
@@ -230,7 +230,7 @@ def _parse_due_date(data: dict, gmt_string) -> datetime:
     """Parse the due date dict into a datetime object."""
     # Add time information to date only strings.
     if len(data["date"]) == 10:
-        data["date"] += "T00:00:00"
+        return datetime.fromisoformat(data["date"]).replace(tzinfo=dt.UTC)
```

Comment:
I think the actual bug is that your timezone is subtracted from a date down the line. Now you're going to return UTC midnight and then it is converted to your local timezone… and then it substracts your timezone again so it ends up at midnight again?

Calendars have all-day events, not sure if our integration has, but that would seem to be the most correct fix.

Reply:
> Now you're going to return UTC midnight and then it is converted to your local timezone

The subtraction is done in https://github.com/home-assistant/core/pull/52258/files#diff-0df90a93485af80ce0879138de9155f75b1a60a63454406902e6246df2da2194L236 . The previous code passed _2021-06-29 00:00:00 +2_ here, which would return _2021-06-28 22:00:00 +0_. Returning here prevents that from happening.